client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/GetWithMetadataOperation.java

```diff
@@ -63,7 +63,7 @@ protected void executeOperation(Channel channel) {
    @Override
    protected void fetchChannelAndInvoke(int retryCount, Set<SocketAddress> failedServers) {
       if (retryCount == 0 && preferredServer != null) {
-         channelFactory.fetchChannelAndInvoke(preferredServer, this);
+         channelFactory.fetchChannelAndInvoke(preferredServer, cacheName(), this);
```

Comment:
Actually looking at the code getWithMetadataAsync has to only go to the exact server specified as well. This is because that server is the one that is aware of the cache operations it has done, so it has to stay as it was before as well.